### PR TITLE
[102X] Use DeepBoosted V02 training

### PR DESCRIPTION
**Pull request by raggleton**
Use V02 training for DeepBoosted classifiers. This training was derived on 94X (2017) MC, and is recommended for everything except 2016v2.

Since this is a fairly important change, I have made this PR against a new base **RunII_102X_v2** branch (which used the head of RunII_102X_v1 as the start).

This PR also requires a newer CMSSW release (gone for the latest 102X), hence changes in install.[c]sh

If in earlier CMSSW release, to avoid shifting releases, can also do:

```
cd $CMSSW_BASE
git cms-merge-topic 26370
mkdir -p $CMSSW_BASE/external/$SCRAM_ARCH/data/RecoBTag/Combined/data/DeepBoostedJet
cp -r /cvmfs/cms.cern.ch/$SCRAM_ARCH/cms/cmssw/CMSSW_10_2_15/external/$SCRAM_ARCH/data/RecoBTag/Combined/data/DeepBoostedJet/V02 $CMSSW_BASE/external/$SCRAM_ARCH/data/RecoBTag/Combined/data/DeepBoostedJet
scram build -j9
```

Note that this will checkout a load of packages, and scram b will take a while longer.

Twiki reference: https://twiki.cern.ch/twiki/bin/view/CMS/DeepAKXTagging

Validation studies currently ongoing. Let's see how the test samples behave - should see a difference in the discriminator values.

Please don't merge yet - I just wanted to start the CI tests rolling in parallel with our own studies.

### CODE REVIEW

**Comment by raggleton** [comment]
Some studies:

tl;dr: V02 has an effect. But discriminator values not identical to 2016v2 (don't know if expected or not). Cut values may need to be different to retain same performance point?

[16_V02_Training_18_9_19.pdf](https://github.com/UHH2/UHH2/files/3627622/16_V02_Training_18_9_19.pdf)

**Comment by raggleton** [comment]
Hmm may need to add:
```
pfMassDecorrelatedDeepBoostedJetTags.min_jet_pt = cms.double(100)
```
to ensure all jets get discriminator (default value is 150, applied to raw jet pT). But do we care about such low pT jets? And are there any even affected?

**Comment by raggleton** [comment]
Interestingly can see some big differences in the deep boosted distribution shapes. 
e.g. https://uhh2-integration.web.cern.ch/UHH2integration/test1426/data_2016v3_JetHT_figs/jetsAk8PuppiSubstructure_SoftDropPuppi.btag_MassDecorrelatedDeepBoosted_probQCDcc.pdf

Extra bonus: due to https://github.com/cms-sw/cmssw/pull/25731, the electron ecalPFClusterIso and hcalPFClusterIso is now fixed
